sig-azure/0018-20180711-azure-availability-zones.md

```diff
@@ -0,0 +1,238 @@
+---
+kep-number: 18
+title: Azure Availability Zones
+authors:
+  - "@feiskyer"
+owning-sig: sig-azure
+participating-sigs:
+  - sig-azure
+  - sig-storage
+reviewers:
+  - name: "@khenidak"
+  - name: "@colemickens"
+approvers:
+  - name: "@brendanburns"
+editor: TBD
+creation-date: 2018-07-11
+last-updated: 2018-07-11
+status: provisional
+---
+
+# Azure Availability Zones
+
+## Table of Contents
+
+- [Azure Availability Zones](#azure-availability-zones)
+  - [Summary](#summary)
+  - [Scopes and Non-scopes](#scopes-and-non-scopes)
+    - [Scopes](#scopes)
+    - [Non-scopes](#non-scopes)
+  - [AZ label format](#az-label-format)
+  - [Cloud provider options](#cloud-provider-options)
+  - [Node registration](#node-registration)
+    - [Get by instance metadata](#get-by-instance-metadata)
+    - [Get by Go SDK](#get-by-go-sdk)
+  - [LoadBalancer and PublicIP](#loadbalancer-and-publicip)
+  - [AzureDisk](#azuredisk)
+    - [PVLabeler](#pvlabeler)
+    - [PersistentVolumeLabel](#persistentvolumelabel)
+    - [StorageClass](#storageclass)
+  - [Appendix](#appendix)
+
+## Summary
+
+This proposal aims to add [Azure Availability Zones (AZ)](https://azure.microsoft.com/en-us/global-infrastructure/availability-zones/) support to Kubernetes.
+
+## Scopes and Non-scopes
+
+### Scopes
+
+The proposal includes required changes to support availability zones for various functions in Azure cloud provider and AzureDisk volumes:
+
+- Detect availability zones automatically when registering new nodes and node's label `failure-domain.beta.kubernetes.io/zone` will be replaced with AZ instead of fault domain
```

Comment:
rather than having the kubelet self-register this, can the node controller pull this information and add it into the node object?